enterprise/utils.py

```diff
@@ -1108,3 +1110,104 @@ def get_platform_logo_url():
         settings.LMS_ROOT_URL,
         get_logo_url()
     ) if get_logo_url else 'http://fake.url'
+
+
+def create_tableau_user(username, enterprise_customer_user):
+    """
+    Create the user in tableau and store the tableau user in the EnterpriseAnalyticsUser model.
+    """
+    from enterprise.models import EnterpriseAnalyticsUser  # pylint: disable=import-outside-toplevel
+    tableau_auth, server = get_tableau_server()
+    if tableau_auth and server and enterprise_customer_user:
+        enterprise_uuid = enterprise_customer_user.enterprise_customer.uuid
+        try:
+            with server.auth.sign_in(tableau_auth):
+                user_item = TSC.UserItem(username, TSC.UserItem.Roles.Viewer)
+                user = server.users.add(user_item)
+                EnterpriseAnalyticsUser.objects.get_or_create(
+                    enterprise_customer_user=enterprise_customer_user,
+                    analytics_user_id=user.id
+                )
+                add_user_to_tableau_group(user.id, str(enterprise_uuid))
+        except ServerResponseError as exc:
+            LOGGER.error(
+                '[TABLEAU USER SYNC] Could not sync enterprise admin user %s in tableau.'
+                'Server returned: %s', username, str(exc),
+            )
```

Comment:
Add a log here in else in case if condition doesn't work

Reply:
Done